lib/services/credentials/credential.dao.js

```diff
@@ -124,7 +124,18 @@ dao.insertCredential = function (id, type, credentialObj) {
 };
 
 dao.getCredential = function (id, type) {
```

Comment:
Maybe this method could be renamed in `getActiveCredential` — so that instead of throwing an error in case it's a `Set`, it could further filter the returned data and return the active credential.